src/nft/utils/pooledAssets.ts

```diff
@@ -44,12 +45,33 @@ export const calcSudoSwapPrice = (asset: GenieAsset, position = 0): string | und
   let currentPrice = BigNumber.from(sudoSwapPool.spotPrice)
   const delta = BigNumber.from(sudoSwapPool.delta)
   const poolFee = BigNumber.from(sudoSwapPool.fee)
+  let virtualTokenBalance = BigNumber.from(sudoSwapPool.spotPrice)
+  let virtualNFTBalance = BigNumber.from(sudoSwapPool.delta)
 
-  for (let i = 0; i <= position; i++) {
+  if (sudoSwapPool.bondingCurve === BondingCurve.Xyk) {
```

Comment:
this function has gotten rather large, do we want to split it into smaller functions for each curve?